### ISSUE TITLE
Vectorization in Factorization model

### ISSUE THREAD

**Issue by campsd** [yhtang/FunFact#70]
Related to #54, #57  

Add full support for vectorization to the factorization model.

Some considerations:

- It probably makes sense to consider _every_ factorization as vectorized even if `nvec=1`
- Bookkeeping of all the best solutions can happen in a copy
- Total loss = sum over vectorized loss function

**Comment by yhtang** [comment]
We discussed earlier today about automatically inferring the vectorization dimensions by shape comparison.
